src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -625,11 +625,8 @@ private struct UrlDecoder
 
             private void FlushBytes()
             {
```

Comment:
You removed the `if (_numBytes > 0)` because the caller already checks for it?  If so, you should add an Assert() here to verify the condition.